pom.xml

```diff
@@ -24,6 +24,7 @@
     <cmake.generate.extra></cmake.generate.extra>
     <cmake.generate.skip>false</cmake.generate.skip>
     <javah.skip>false</javah.skip>
+    <maven.assembly.id>${os.detected.name}-${os.detected.arch}-${os.detected.bitness}</maven.assembly.id>
```

Comment:
@bmhm there seems to be an odd bug when we use a plugin-generated property such as `${os.detected.name}` from within an assembly.  This bug is not reproducible if we make up our own property.
* One option is to use our own replacement properties throughout
* Another option is to simply declare a single property to be picked up later (That's the one I've chosen)
   * I didn't know if the property name is kosher.  Please rename as needed.

Variable resolution in Maven is unlike any system I've used before.  From what I've seen it does forward lookups , which can be both convenient as well as a pain in the ass.  For example, `<foo.bar>${foo.bar}</foo.bar>` is illegal, which is quite cumbersome as it prevents appending stuff (such as `cmake` flags) to an already existing variable.  I believe this is due to the nature of profiles, but anyway... This above fix seems to work.

Reply:
That is what I also wanted to propose, so yes, go ahead. We can file an upstream bug, if you wish.

The property name won’t clash. The closest I could find would be `assembly.appendAssemblyId`. So I’ll just mark this as resolved. :)